google-cloud-spanner/src/main/java/com/google/cloud/spanner/GrpcStreamIterator.java

```diff
@@ -182,4 +191,9 @@ public boolean cancelQueryWhenClientIsClosed() {
       return this.cancelQueryWhenClientIsClosed;
     }
   }
+
+  private void onStreamMessage(PartialResultSet partialResultSet) {
+    Optional.ofNullable(streamMessageListener)
+        .ifPresent(sl -> sl.onStreamMessage(partialResultSet, prefetchChunks >= stream.size()));
```

Comment:
Is `prefetchChunks >= stream.size()` correct for `bufferIsFull`? Would it not make more sense to use `stream.remainingCapacity() <= 1`? (Assuming that we do not double the buffer size)

Then we could also remove `this.prefetchChunks` as a field.

Reply:
done